### PR TITLE
Periodical evaluation sidepanel navigation - RISDEV-4749

### PR DESCRIPTION
**Pull request by HPrinz**

### CODE REVIEW

File: frontend/src/domain/legalPeriodicalEdition.ts [hunk 1]
```diff
@@ -21,9 +21,17 @@ export default class LegalPeriodicalEdition {
 
   constructor(data: Partial<LegalPeriodicalEdition> = {}) {
     Object.assign(this, data)
+    if (this.id === undefined) {
+      this.id = crypto.randomUUID()
+    }
   }
 
-  get missingRequiredFields() {
```
Comment (Nadav-B) [inline frontend/src/domain/legalPeriodicalEdition.ts:26]: Method naming: when defining a getter in js / ts, there is no need to use the get prefix in the method name itself. I renamed it on this [commit](https://github.com/digitalservicebund/ris-backend-service/commit/8ba4d1e0f241e89dac90bc6e52b683499ca47b5b)

File: frontend/src/stores/editionStore.ts [hunk 2]
```diff
@@ -14,13 +14,13 @@ export const useEditionStore = defineStore("editionStore", () => {
...
   > {
-    const uuid = route.params.uuid
+    const uuid = route.params.editionId
     const response = await LegalPeriodicalEditionService.get(
       uuid.toString() as UUID,
     )
 
     if (response.data) {
-      edition.value = response.data
+      edition.value = response.data as LegalPeriodicalEdition
```
Comment (Nadav-B) [inline frontend/src/stores/editionStore.ts:23]: Unfortunately, this `as` won't work, as it does not how to parse the java object into a typescript object. We need to call the constructor. I changed it in the [commit](https://github.com/digitalservicebund/ris-backend-service/commit/8ba4d1e0f241e89dac90bc6e52b683499ca47b5b#diff-2de3faa42cad092a3f1452f96bfbadca9b008ff43ebc477d8f0922a385c7d7bd). It was the cause why the missing method could not be called. When we use the constructor once on fetching level, we don't need it anymore in component levels

File: frontend/test/e2e/caselaw/legal-periodical-evaluation.spec.ts [hunk 3]
```diff
@@ -156,13 +143,33 @@ test.describe(
...
+          await expect(page.getByText("Pflichtfeld nicht befüllt")).toBeHidden()
+          // TODO should be empty from the beginning
+          await clearInput(page, "Name der Ausgabe")
+          await page.getByLabel("Speichern").click()
           await expect(
             page.locator(`text="Pflichtfeld nicht befüllt"`),
           ).toHaveCount(1)
         })
 
+        // await test.step("prefix or name have to be chosen", async () => {
```
Comment (Nadav-B) [inline frontend/test/e2e/caselaw/legal-periodical-evaluation.spec.ts:166]: This line can be removed as it deprecated to our current validation

Comment (HPrinz) [reply frontend/test/e2e/caselaw/legal-periodical-evaluation.spec.ts:166]: Thank you! Done ✅

Comment (HPrinz) [reply frontend/test/e2e/caselaw/legal-periodical-evaluation.spec.ts:166]: Done ✅

File: frontend/src/stores/editionStore.ts (hunk 2 above)
Comment (HPrinz) [reply frontend/src/stores/editionStore.ts:23]: Thank you so much! ❤️

**Pull request merged**